packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt

```diff
@@ -0,0 +1,268 @@
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ActivityEventListener
+import org.unimodules.core.interfaces.ActivityProvider
+import org.unimodules.core.interfaces.ExpoMethod
+import org.unimodules.core.interfaces.services.UIManager
+import java.util.*
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+
+class LocalAuthenticationModule(context: Context) : ExportedModule(context), ActivityEventListener {
+  private val AUTHENTICATION_TYPE_FINGERPRINT = 1
+  private val AUTHENTICATION_TYPE_FACIAL_RECOGNITION = 2
+  private val AUTHENTICATION_TYPE_IRIS = 3
+  private val SECURITY_LEVEL_NONE = 0
+  private val SECURITY_LEVEL_SECRET = 1
+  private val SECURITY_LEVEL_BIOMETRIC = 2
+  private val biometricManager = BiometricManager.from(context)
+  private val packageManager = context.packageManager
+  private var biometricPrompt: BiometricPrompt? = null
+  private var promise: Promise? = null
+  private var isAuthenticating = false
+  private lateinit var moduleRegistry: ModuleRegistry
+  private var uIManager: UIManager? = null
+
+  private fun convertErrorCode(code: Int): String {
+    return when (code) {
+      BiometricPrompt.ERROR_CANCELED, BiometricPrompt.ERROR_NEGATIVE_BUTTON, BiometricPrompt.ERROR_USER_CANCELED -> "user_cancel"
+      BiometricPrompt.ERROR_HW_NOT_PRESENT, BiometricPrompt.ERROR_HW_UNAVAILABLE, BiometricPrompt.ERROR_NO_BIOMETRICS, BiometricPrompt.ERROR_NO_DEVICE_CREDENTIAL -> "not_available"
+      BiometricPrompt.ERROR_LOCKOUT, BiometricPrompt.ERROR_LOCKOUT_PERMANENT -> "lockout"
+      BiometricPrompt.ERROR_NO_SPACE -> "no_space"
+      BiometricPrompt.ERROR_TIMEOUT -> "timeout"
+      BiometricPrompt.ERROR_UNABLE_TO_PROCESS -> "unable_to_process"
+      else -> "unknown"
+    }
+  }
+
+  private val authenticationCallback: BiometricPrompt.AuthenticationCallback = object : BiometricPrompt.AuthenticationCallback() {
+    override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+      isAuthenticating = false
+      biometricPrompt = null
+      promise?.resolve(Bundle().apply {
+        putBoolean("success", true)
+      })
+      promise = null
+    }
+
+    override fun onAuthenticationError(errMsgId: Int, errString: CharSequence) {
+      isAuthenticating = false
+      biometricPrompt = null
+      promise?.resolve(Bundle().apply {
+        putBoolean("success", false)
+        putString("error", convertErrorCode(errMsgId))
+        putString("message", errString.toString())
+      })
+      promise = null
+    }
+  }
+
+  override fun getName(): String {
+    return "ExpoLocalAuthentication"
+  }
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    this.moduleRegistry = moduleRegistry
+    uIManager = moduleRegistry.getModule(UIManager::class.java)
+    uIManager?.registerActivityEventListener(this)
```

Comment:
To get things from the module registry you can use a property delegation. See https://github.com/expo/expo/blob/09cfa47b77a4391c20c46385d508ffaeaf50c590/packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt#L58. It requires a couple more set up instructions which can be found in the ImagePicker module.